src/librustc_mir_build/build/matches/test.rs

```diff
@@ -414,7 +414,40 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
 
         let deref_ty = match ty.kind {
             ty::Ref(_, deref_ty, _) => deref_ty,
-            _ => bug!("non_scalar_compare called on non-reference type: {}", ty),
```

Comment:
Isn't this expecting a reference because the caller of `non_scalar_compare` *borrowed* the values being compared?

cc @nikomatsakis @matthewjasper I'm unclear on what the intended use of `non_scalar_compare` is.

Reply:
most (all?) of the time it was because the caller already had a reference to work with.